src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -993,12 +1007,41 @@ static ImmutableArray<Symbol> getAllTypeAndRequiredMembers(TypeSymbol containing
                             foreach (var (_, requiredMember) in requiredMembers)
                             {
                                 // We want to assume that all required members were _not_ set by the chained constructor
-                                builder.Add(getFieldSymbolToBeInitialized(requiredMember));
+                                builder.AddRange(getAllMembersToBeDefaulted(requiredMember));
                             }
 
                             return builder.ToImmutableAndFree();
                         }
 
+                        static ImmutableArray<Symbol> getAllMembersToBeDefaulted(Symbol requiredMember)
+                        {
+                            Debug.Assert(requiredMember.IsRequired());
+
+                            if (requiredMember is FieldSymbol)
+                            {
+                                return ImmutableArray.Create<Symbol>(getFieldSymbolToBeInitialized(requiredMember));
```

Comment:
It looks like in this case `getFieldSymbolToBeInitialized` always returns the input. Is there a specific reason for doing this call here?

Reply:
Consistency, but you're right that we'll probably never change the field symbol returned here :).